docs/knuu-new.md

```diff
@@ -0,0 +1,161 @@
+
+## Table of Contents
+
+- [Installation](#installation)
+- [Creating a `Knuu` Object](#creating-a-knuu-object)
+- [Handling Stop Signals](#handling-stop-signals)
+- [Cleaning Up Resources](#cleaning-up-resources)
+
+## Installation
+
+To use the `knuu` package, ensure you have the necessary dependencies installed:
+
+```bash
+go get github.com/celestiaorg/knuu
+```
+
+## Creating a `Knuu` Object
+
+To create a `Knuu` object, you need to provide several options encapsulated in the `Options` struct. The object is initialized with a Kubernetes client, MinIO client, image builder, and other configurations.
+To use the default knuu object, an empty `Options` struct can be passed to the `New` function.
+
+### Options
+
+- `K8sClient`: A Kubernetes client used to interact with your Kubernetes cluster (_default: k8s config from the running environment i.e. kubeconfig file_).
+- `MinioClient`: A custom MinIO client for managing object storage (_default: nil_).
+- `ImageBuilder`: A custom builder for creating container images (_default: kaniko builder_).
+- `Scope`: A unique identifier for the resources managed by this knuu object (_default: a pseudo random string_).
+- `ProxyEnabled`: A boolean to enable or disable a reverse proxy (_default: false_).
+- `Timeout`: Duration after which the resources will be automatically cleaned up (_default: 60 minutes_).
+- `Logger`: A logger instance for logging (_default: logrus with info level_).
+
+### Example
+
+```go
+package main
+
+import (
+    "context"
+    "log"
+    "time"
+
+    "github.com/celestiaorg/knuu/pkg/k8s"
+    "github.com/celestiaorg/knuu/pkg/minio"
+    "github.com/celestiaorg/knuu/pkg/builder"
+    "github.com/sirupsen/logrus"
+    "github.com/celestiaorg/knuu"
+)
+
+func main() {
+    ctx := context.Background()
+
+    options := knuu.Options{
+        Scope:        "example-scope",
+        Timeout:      30 * time.Minute,
+        Logger:       logrus.New().WithLevel(logrus.ErrorLevel),
+    }
+
+    kn, err := knuu.New(ctx, options)
+    if err != nil {
+        log.Fatalf("Failed to create Knuu object: %v", err)
+    }
+    kn.HandleStopSignal(ctx) // explained in the next section
+
+    defer func() {
+        err := kn.CleanUp(ctx)
+        if err != nil {
+            log.Fatalf("Failed to clean up resources: %v", err)
+        }
+    }()
+
+
+    sampleInstance, err := kn.NewInstance("alpine")
+    if err != nil {
+        log.Fatalf("Failed to create instance: %v", err)
+    }
+
+    err = sampleInstance.Build().SetGitRepo(ctx, builder.GitContext{
+		Repo:     "https://github.com/<sample-repo>.git",
+		Branch:   "<desired-branch>",
+		Username: "<git-username>", // Leave empty if repo is public
+		Password: "<git-password>", // Leave empty if repo is public
+	})
```

Comment:
Can we add some comments to explain this more? 

- Why alpine? Does this need to related to the git repo at all? 
- Is it assumed that the git repo has a dockerfile?

Reply:
ok I will add more comments

Reply:
BTW, the alpine is just a name. I will rename it to something else to avoid confusion. The git builder will follow whatever image is defined in the Dockerfile in the given repo.